node/core/dispute-coordinator/src/participation/queues/mod.rs

```diff
@@ -160,6 +160,21 @@ impl PartialEq for ParticipationRequest {
 }
 #[cfg(test)]
 impl Eq for ParticipationRequest {}
+#[cfg(test)]
+impl Clone for ParticipationRequest {
```

Comment:
I would just not have implemented `Clone` at all. This is a bit hacky. Instead of cloning, we should be able to create new instances in tests where necessary. E.g. just a new `let ... = create_test_partitipation_request` ... instead of cloning the previous one.

Reply:
I went ahead and tried your suggestion. It turns out there's some non-determinism in the helper `make_participation_request()` such that the candidate hashes end up different. Particularly the `signature` field in the `CandidateDescriptor` within the `CandidateReceipt` is different each time it is generated.

This means that calling `make_participation_request()` multiple times results in requests that don't compare as equal, ruining our asserts.

Know of other ways we can do this without clone?

Reply:
Or here's another idea which might be considered better form. 

I could at least move the `clone` functionality into `tests.rs` as a helper function rather than implementing it on the `ParticipationRequest` type.

Reply:
yes, making it a standalone function like `test_clone` core something would also be way better. The non-determinism is interesting, I don't see where it would be coming [from](https://github.com/paritytech/polkadot/blob/a8e6828115f852dc08a90c9fc0698f59a589b977/node/core/dispute-coordinator/src/participation/queues/tests.rs#L25)?